tests/unit/test_args.py

```diff
@@ -125,3 +126,31 @@ def test_max_pool_from_client_config_forwarded_to_endpoint_creator(self):
                 response_parser_factory=None, timeout=(60, 60), verify=True,
                 max_pool_connections=20
             )
+
+
+class TestExecutionEnvAppend(unittest.TestCase):
+    def setUp(self):
+        self.args_create = args.ClientArgsCreator(mock.Mock(),
+                                                  'user-agent prefix',
+                                                  None, None)
+
+    def test_execution_env_append(self):
+        with mock.patch('os.environ.get') as m:
+            m.return_value = 'FooEnv'
+            result = self.args_create.compute_client_args(mock.MagicMock(),
+                                                          None,
+                                                          mock.MagicMock(),
+                                                          'us-west-2',
+                                                          '', False, {})
+            self.assertNotEqual(re.match('.+ exec-env/FooEnv$',
```

Comment:
I would prefer not to use regex. You can use `.endswith` to achieve what you want.